.github/workflows/ci.yml

```diff
@@ -161,3 +161,24 @@ jobs:
           alert-threshold: '5%'
           comment-always: true
           fail-on-alert: false
+
+  clang-tidy:
+    runs-on: ubuntu-latest
+    env:
+      CC: clang
+      CXX: clang++
+    steps:
+      - uses: actions/checkout@v4
+      - run: cmake --version
+      - name: Install llvm
+        run: sudo bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)"
```

Comment:
Why do we need this? I thought we tweaked the CMake scripts to conveniently download ClangTidy binaries when configuring?